src/bitcoin-chainstate.cpp

```diff
@@ -64,10 +64,10 @@ int main(int argc, char* argv[])
 
     // SETUP: Context
     kernel::Context kernel_context{};
-    // We can't use a goto here, but we can use an assert since none of the
-    // things instantiated so far requires running the epilogue to be torn down
-    // properly
-    assert(kernel::SanityChecks(kernel_context));
+    if (!kernel::SanityChecks(kernel_context)) {
+        std::cerr << "Failed sanity check.";
+        return 0;
```

Comment:
nit: this being a console app, it might make sense to return a non-zero error code when `cerr` is used

Reply:
Yeah, done.